### PR TITLE
has and snapshot specs

### PR DESCRIPTION
**Pull request by AlexBHarley**

### CODE REVIEW

**Comment by yasserf** [comment]
Is this missing has or am I going blind? =D

**Comment by AlexBHarley** [comment]
These are just the ones copied over from the java client. Probably need own file for has with different scenarios. There is one scenario in this file at the bottom

**Comment by yasserf** [comment]
Yup, a second file would be awesome. 

Otherwise all the PRs looked great and have been merged!

File: record/record-has.feature [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		And the server sends the message C|A+
+		And the client logs in with username "XXX" and password "YYY"
+		And the server sends the message A|A+
+
+    # The client creates a record
+	#
+
+	# The client queries for an existing record
+	Given the client checks if the server has the record "existingRecord"
+    When the server sends the message R|H|existingRecord|T|+
```
Comment (yasserf) [inline record/record-has.feature:20]: Extra message part at end?

File: record/record-has.feature [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	# The client is connected
+	Given the test server is ready
+		And the client is initialised
+		And the server sends the message C|A+
+		And the client logs in with username "XXX" and password "YYY"
+		And the server sends the message A|A+
+
+    # The client creates a record
+	#
```
Comment (yasserf) [inline record/record-has.feature:16]: Creates record? Specs don't need this!:D

File: record/record-snapshot.feature [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	# The client is connected
+	Given the test server is ready
+		And the client is initialised
+		And the server sends the message C|A+
+		And the client logs in with username "XXX" and password "YYY"
+		And the server sends the message A|A+
+
+    # The client creates a record, but doesn't get a read
+	Given the client creates a record named "snapshotRecord"
```
Comment (yasserf) [inline record/record-snapshot.feature:18]: Not needed

**Pull request merged**